clang/test/CodeGenSYCL/Inputs/sycl.hpp

```diff
@@ -187,6 +187,7 @@ class accessor {
 template <int dimensions, access::mode accessmode, access::target accesstarget>
 struct opencl_image_type;
 
+#ifdef __SYCL_DEVICE_ONLY__
```

Comment:
Note the CFE only provides these types in device mode, so this change lets us uses this in header properly in host mode.